### DISCUSSION TITLE
Should recipe ingredients be scraped?

### DISCUSSION THREAD

**Discussion by schmitzkr** [TomBursch/kitchenowl#474] [Q&A]
Hi there, fantastic tool, thanks for all your hard work.  I am trying to move away from AnyList to a selfhosted option, KitchenOwl seems to be meeting my requirements for shopping list/recipe.  One thing I noticed was when I add a recipe the ingredients are not added.  I am wondering if this is just coincidence, (all the links I have given it are not able to pull the ingredients).  Or scraping those ingredients is not  implemented?  It would be very useful to be able to show ingredients in an imported recipe without having to manually add them
I noticed that there are tools for scraping that data (https://github.com/hhursev/recipe-scrapers).

**Comment by TomBursch** [comment]
Hi, thanks! Yes, the library you linked is what we use for recipe scraping. The problem is that the ingredients are returned as strings (e.g. `5 table spoons of Sugar`), and it's not easy to split the description from the item name and match it to a KitchenOwl item from the database. It should work for some items, but is a bit inconsistent right now. Take a look at https://github.com/TomBursch/kitchenowl/issues/163 too.

**Reply by schmitzkr** [reply]
Hey, thanks for the reply!  Thank you very much for explaining the inconsistent part.   I guess making this more consistent is only in the control of those library maintainers then.
Again thanks for all the hard work on this project!
